crates/bitwarden/src/auth/login/api_key.rs

```diff
@@ -28,24 +26,26 @@ pub(crate) async fn api_key_login(
     let response = request_api_identity_tokens(client, input).await?;
 
     if let IdentityTokenResponse::Authenticated(r) = &response {
-        client.set_tokens(
-            r.access_token.clone(),
-            r.refresh_token.clone(),
-            r.expires_in,
-            LoginMethod::ApiKey {
-                client_id: input.client_id.to_owned(),
-                client_secret: input.client_secret.to_owned(),
-            },
-        );
-
         let access_token_obj = decode_token(&r.access_token)?;
 
         // This should always be Some() when logging in with an api key
         let email = access_token_obj
             .email
             .ok_or(Error::Internal("Access token doesn't contain email"))?;
 
-        let _ = determine_password_hash(client, &email, &input.password).await?;
+        let kdf = request_prelogin(client, email.clone()).await?.try_into()?;
```

Comment:
```suggestion
        let kdf = client.prelogin(email.clone()).await?;
```

Reply:
Long term I would like to eliminate the explicit consumer prelogin call. Maybe that results in us needing an AuthBuilder or something.